docs/adr/ADR-001-Theme_and_tokens.md

```diff
@@ -0,0 +1,87 @@
+# ADR-001: Theme and tokens
+
+## Date
+
+2023-10-04
+
+## Status
+
+PROPOSED
+
+## Context
+
+At the moment, when providing custom themes to the Design System, the user can only override the palette of colors.
+In some cases, this can be limiting, as the user may need to override specific parts of components to respect A11Y
+guidelines or to have more coherent colors.
+
+## Decision
+
+We will change the structure of the Theme object provided to the Design System to allow providing overrides for either
+the palette or component tokens.
+Any change in the palette will be reflected in the component tokens.
+Component tokens will accept only colors from the palette.
+At the moment, only colors can be overridden, but this could be extended to other tokens in the future.
+
+The input structure accepted by the `DesignSystem` provider would be as follows:
+
+```typescript
+interface ThemeCustomization {
+    palette: {
+        brand-<tint>: string; // e.g: brand-05: #E0F0F9
+        accent-<tint>: string;
+        neutral-<tint>: string;
+        alert-<tint>: string;
+        informative-<tint>: string;
+        success-<tint>: string;
+        warning-<tint>: string;
+    };
+    components: {
+        <component>-<flavor?>-<property>: <palette-token>;
+        button-primary-background-color: 'brand-05';
+        ...
+    };
+}
+```
+
+The output structure exposed by the `useTheme` hook and the `ThemeProvider` would be as follows:
+
+```typescript
+interface Theme {
+    ref: {
+        palette: Palette; // This should be a shared structure with ThemeCustomization['palette']  
+    };
+    components: ComponentTokens; // This should be a shared structure with ThemeCustomization['components']
```

Comment:
Not sure but should `components` be singular if `ref` is?

Reply:
Je pense que je le voyais comme `reference tokens` et `components tokens`, même si l'interface je l'avais mis sans `s` 😅